units/pod_creation.go

```diff
@@ -178,3 +185,43 @@ func (j *podCreationJob) populateIfUnset(ctx context.Context) error {
 
 	return nil
 }
+
+func (j *podCreationJob) logTaskTimingStats() error {
+	if j.pod.Type != pod.TypeAgent {
+		return nil
+	}
+
+	disp, err := dispatcher.FindOneByPodID(j.pod.ID)
+	if err != nil {
+		return errors.Wrap(err, "finding dispatcher for task")
+	}
+	if disp == nil {
+		return errors.Errorf("dispatcher with pod '%s' not found", j.pod.ID)
+	}
+
+	msg := message.Fields{
+		"message":          "created pod to run container tasks",
+		"pod":              j.pod.ID,
+		"dispatcher_group": disp.GroupID,
+	}
+
+	tsk, err := task.FindOneId(disp.GroupID)
+	if err != nil {
+		return errors.Wrapf(err, "finding tasks associated with dispatcher group '%s'", disp.GroupID)
+	}
+
+	if tsk != nil {
+		msg["is_for_task_group"] = false
```

Comment:
I want to keep the if-else because of the comment on the else case, but it would look more awkward to add an if-else block whose sole purpose is to no-op and make a comment.